#!/usr/bin/env python
# -*- coding: utf-8 -*-
PACKAGE = "robot_vision"
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("h_lower", int_t, 0, "HSV color space h_low", 0, 0, 255)
gen.add("s_lower", int_t, 0, "HSV color space s_low", 0, 0, 255)
gen.add("v_lower", int_t, 0, "HSV color space v_low", 0, 0, 255)
gen.add("h_upper", int_t, 0, "HSV color space h_high", 255, 0, 255)
gen.add("s_upper", int_t, 0, "HSV color space s_high", 255, 0, 255)
gen.add("v_upper", int_t, 0, "HSV color space v_high", 255, 0, 255)

gen.add("row_1", int_t, 0, "红绿灯识别区域上方", 0, 0, 479)
gen.add("row_2", int_t, 0, "红绿灯识别区域下方", 120, 0, 479)
gen.add("col_1", int_t, 0, "红绿灯识别区域左侧", 0, 0, 639)
gen.add("col_2", int_t, 0, "红绿灯识别区域右侧", 639, 0, 639)

exit(gen.generate(PACKAGE, "robot_vision", "light_hsv"))
